build/rewrite.go

```diff
@@ -54,13 +58,33 @@ func allowedSort(name string) bool {
 	return false
 }
 
-// Rewrite applies the high-level Buildifier rewrites to f, modifying it in place.
+// Rewrites controls the rewrites to be applied.
+//
+// If non-nil, the rewrites with the specified names will be run. If
+// nil, a default set of rewrites will be used that is determined by
+// the type (BUILD vs default starlark) of the file being rewritten.
+type Rewriter struct {
+	DisableRewrites                 *[]string
+	IsLabelArg                      *map[string]bool
+	LabelDenyList                   *map[string]bool
+	IsSortableListArg               *map[string]bool
+	SortableDenylist                *map[string]bool
+	SortableAllowlist               *map[string]bool
+	NamePriority                    *map[string]int
+	StripLabelLeadingSlashes        *bool
+	ShortenAbsoluteLabelsToRelative *bool
+}
+
 func Rewrite(f *File) {
+	var rewriter = &Rewriter{}
+	rewriter.Rewrite(f)
+}
+
+func (w *Rewriter) Rewrite(f *File) {
 	for _, r := range rewrites {
-		if !disabled(r.name) {
-			if f.Type&r.scope != 0 {
-				r.fn(f)
-			}
+		// f.Type&r.scope is a bitwise comparison
+		if !disabled(w, r.name) && (f.Type&r.scope != 0) || (w.DisableRewrites != nil && !disabled(w, r.name)) {
```

Comment:
Explanation of this if statement - We require the conditional or here because (f.Type&r.scope != 0) is intended ot check the scope. For starlark files, the scope tells us not to do call sort which is why we need to have our own check on the right side.

Reply:
Fair enough. Would add that as a comment in the code.